src/private/inventory.ts

```diff
@@ -29,6 +30,52 @@ interface GetItemRequest {
     instance_number: number;
 }
 
+const InventoryItemSchema = yup.object({
+    inventory_address: yup
+        .string()
+        .strict()
+        .typeError("inventory_address must be a string.")
+        .required("The inventory_address (string) must be provided."),
+    item_id: yup
+        .number()
+        .typeError("item_id must be a number.")
+        .positive("item_id must be a positive number.")
+        .required("The item_id (integer) must be provided."),
+    instance_number: yup
+        .number()
+        .typeError("instance_number must be a number.")
+        .positive("instance number must be a positive number.")
+        .required("The instance_number (integer) must be provided.")
+});
+
+const Data = yup.object({
```

Comment:
oops sorry, that's not what I meant. 

You could have a `dataValidator` var that you just stick inside the other schemas that have a Data field:

```
const dataValidator = yup.lazy((value) => {  if .... })
```

and in InventoryItemSchema:

```
const InventoryItemSchema = yup.object({
   ...
  data: dataValidator
   ...
})
```

Instead of having to call 2 different validators